hls-test-utils/src/Development/IDE/Test/Diagnostic.hs

```diff
@@ -32,6 +33,17 @@ requireDiagnostic actuals expected@(severity, cursor, expectedMsg, expectedTag)
         && standardizeQuotes (T.toLower expectedMsg) `T.isInfixOf`
            standardizeQuotes (T.toLower $ d ^. message)
         && hasTag expectedTag (d ^. tags)
+        && codeMatches d
+
+    codeMatches d =
+#if MIN_VERSION_GLASGOW_HASKELL(9,5,0,0)
+        case (mbExpectedCode, _code d) of
+          (Nothing, _)                         -> True
+          (Just expectedCode, Nothing)         -> False
+          (Just expectedCode, Just actualCode) -> InR expectedCode == actualCode
+#else
+        True
```

Comment:
Could we get away with avoiding CPP here? 
`Compat.hs` defines:
```
data GhcVersion
  = GHC94
  | GHC96
  | GHC98
  | GHC910
  deriving (Eq, Ord, Show, Enum)
  ```
  
 which could turn this CPP into a run-time check like:
 ```suggestion
    codeMatches d
      | ghcVersion >= GHC96 =
          case (mbExpectedCode, _code d) of
            (Nothing, _)                         -> True
            (Just expectedCode, Nothing)         -> False
            (Just expectedCode, Just actualCode) -> InR expectedCode == actualCode
      | otherwise = True
```
 
Unless I am overlooking something

Reply:
You're right, this was me being lazy and not wanting to figure out how to import `ghcVersion`. My bad.